Gems/SceneProcessing/Code/Source/Generation/Components/TangentGenerator/TangentGenerateComponent.cpp

```diff
@@ -28,11 +28,15 @@
 #include <SceneAPI/SceneData/GraphData/MeshVertexTangentData.h>
 
 #include <AzCore/Math/Vector4.h>
+#include <AzCore/Settings/SettingsRegistry.h>
 #include <AzCore/std/smart_ptr/make_shared.h>
 
 
 namespace AZ::SceneGenerationComponents
 {
+    static constexpr const char* DefaultTangentGenerationKey{ "/O3DE/SceneAPI/TangentGenerateComponent/DefaultGenerationMethod" };
+    static constexpr const char* DebugBitangentFlipKey{ "/O3DE/SceneAPI/TangentGenerateComponent/DebugBitangentFlip" };
```

Comment:
These are being passed to `SettingsRegistry::Get()`, which takes a `AZStd::string_view` for the key. Why not have these statics be `string_view`s directly?
```suggestion
    static constexpr AZStd::string_view DefaultTangentGenerationKey{ "/O3DE/SceneAPI/TangentGenerateComponent/DefaultGenerationMethod" };
    static constexpr AZStd::string_view DebugBitangentFlipKey{ "/O3DE/SceneAPI/TangentGenerateComponent/DebugBitangentFlip" };
```

Reply:
Would this be guaranteed to be null-terminated, such that you could safely use DefaultTangentGenerationKey.data() for functions expecting a char *, like in AZ_Warning? Otherwise you have to construct a string, then use .c_str().

I was following another example from the scene settings, but string_view does appear to be the predominant use across the engine for these keys.

Reply:
To pass the `string_view` to a string formatting function, you use the `AZ_STRING_FORMAT` macro as the placeholder in the format string (which expands to `%.*s`), and the `AZ_STRING_ARG` macro for the argument (which expands to `aznumeric_cast<int>(str.size()), str.data()`. So for example:
```cpp
AZ_Warning("No value found for key " AZ_STRING_FORMAT "!", AZ_STRING_ARG(DefaultTangentGenerationKey));
```

Reply:
Done.